dsp-user/repo/src/test/scala/dsp/user/repo/impl/UserRepoImplSpec.scala

```diff
@@ -23,65 +26,81 @@ object UserRepoImplSpec extends ZIOSpecDefault {
-    username   <- Username.make("username2")
-    email      <- Email.make("email2@email.com")
-    password   <- Password.make("password2")
-    language   <- LanguageCode.make("en")
-    user = User.make(
-             givenName,
-             familyName,
-             username,
-             email,
-             password,
-             language
-           )
-  } yield (user)).toZIO
+  private val user1: User = SharedTestData.simpleUser1
+  private val user2: User = SharedTestData.simpleUser2
 
   val userTests =
-    test("store a user and retrieve by ID") {
+    test("store several users and retrieve all") {
       for {
-        user          <- testUser1
-        _             <- UserRepo.storeUser(user)
-        retrievedUser <- UserRepo.getUserById(user.id)
-      } yield assertTrue(retrievedUser == user)
+        _              <- UserRepo.storeUser(user1)
+        _              <- UserRepo.storeUser(user2)
+        retrievedUsers <- UserRepo.getUsers()
+      } yield assertTrue(retrievedUsers.size == 2)
     } +
-      test("retrieve the user by username") {
+      test("store a user and retrieve by ID") {
+        for {
+          _             <- UserRepo.storeUser(user1)
+          retrievedUser <- UserRepo.getUserById(user1.id)
+        } yield assertTrue(retrievedUser == user1)
+      } +
+      test("store a user and retrieve the user by username") {
         for {
-          user          <- testUser1
-          _             <- UserRepo.storeUser(user)
-          retrievedUser <- UserRepo.getUserByUsernameOrEmail(user.username.value)
-        } yield assertTrue(retrievedUser == user)
+          _             <- UserRepo.storeUser(user1)
+          retrievedUser <- UserRepo.getUserByUsername(user1.username)
+        } yield assertTrue(retrievedUser == user1)
       } +
-      test("retrieve the user by email") {
+      test("store a user and retrieve the user by email") {
         for {
-          user1         <- testUser1
-          user2         <- testUser2
           _             <- UserRepo.storeUser(user1)
-          retrievedUser <- UserRepo.getUserByUsernameOrEmail(user1.email.value)
+          retrievedUser <- UserRepo.getUserByEmail(user1.email)
         } yield {
           assertTrue(retrievedUser == user1) &&
           assertTrue(retrievedUser != user2)
         }
+      } +
+      test("return None if a username already exists") {
+        for {
+          _     <- UserRepo.storeUser(user1)
+          error <- UserRepo.checkIfUsernameExists(user1.username).exit
+        } yield assert(error)(
+          fails(equalTo(None))
+        )
+      } +
+      test("return success (Unit) if a username is unique") {
+        for {
+          _      <- UserRepo.storeUser(user1)
+          result <- UserRepo.checkIfUsernameExists(SharedTestData.simpleUser2.username)
+        } yield assertTrue(result == ())
+      } +
+      test("return None if an email already exists") {
```

Comment:
the description of this test is not wrong, but maybe it would be more clear how it's being tested, if the description would say that the `None` is returned in the error channel. (As you do in the tests where you write `return success (Unit)`)